include/onnxruntime/core/platform/EigenNonBlockingThreadPool.h

```diff
@@ -1465,12 +1469,14 @@ int CurrentThreadId() const final {
       Task t = q.PopFront();
       if (!t) {
         // Spin waiting for work.
-        for (int i = 0; i < spin_count && !t && !done_; i++) {
+        for (int i = 0; i < spin_count && !done_; i++) {
           if (((i+1)%steal_count == 0)) {
             t = Steal(StealAttemptKind::TRY_ONE);
           } else {
             t = q.PopFront();
           }
+          if (t) break;
```

Comment:
>if (t) break; [](http://example.com/codeflow?start=10&length=13)

legit but assemble conditions in "for" is more likely for compiler to have optimized switch.

Reply:
this is not my change but I guess the compiler optimization won't pass memory barriers.